### PR TITLE
Change download directory setting UX

### LINKED ISSUE #437: Changing download directory does not create it if necessary
On Kiwix Desktop 2.0.1 with Linux Mint 19.3 whenever I try to change the download directory (F12 -> Download directory) a alert dialog with the message `Invalid download path` appears.
image

### PR DESCRIPTION
**Pull request by jetownfeve21**
- remove "apply" button
- kiwixPort and zoomFactor are saved onchange
- download directory has a confirm dialog

fix #437

### CODE REVIEW

**Comment by jetownfeve21** [comment]
I cannot reproduce the case on my computer, @mgautierfr do you have this crash ?

**Comment by mgautierfr** [comment]
@kelson42 I cannot reproduce your bug.

If you agree with the change, you can merge this PR.

**Comment by kelson42** [comment]
@mgautierfr I have achieved to reproduced a few days ago, let me try again.

**Comment by kelson42** [comment]
This is the result of the last testing:
image

I believe it would be better if:
* The path on the left
* Considering the box is read-only, no border

I was not able to reproduce the crash but had an other problem:
* Create new directory /tmp/4242
* Chosing it
* Dialog displays properly: "The new download directory path will be: /tmp/4242"
* So far so good
* Now trying to choose "/tmp", I get no error but the directory is not updated from "/tmp/4242" to "/tmp".

**Comment by mgautierfr** [comment]
> The path on the left
Considering the box is read-only, no border

Ok

> Now trying to choose "/tmp", I get no error but the directory is not updated from "/tmp/4242" to "/tmp".

Be sure that you don't just go to the parent directory in the browser. If you do just that, the directory "4242" is still selected (even if you are not "in it").

**Comment by kelson42** [comment]
> > Now trying to choose "/tmp", I get no error but the directory is not updated from "/tmp/4242" to "/tmp".
> 
> Be sure that you don't just go to the parent directory in the browser. If you do just that, the directory "4242" is still selected (even if you are not "in it").

Indeed, so we have only the styling problem.

**Comment by mgautierfr** [comment]
Should be good now.
